addons/address.py

```diff
@@ -156,21 +156,21 @@ def _inet_address_convert_to_cidr(self, ifaceobjlist):
             # If user address is not in CIDR notation, convert them to CIDR
             for addr_index in range(0, len(addrs)):
                 addr = addrs[addr_index]
+                newaddr = addr
                 if '/' in addr:
                     newaddrs.append(addr)
-                    continue
-                newaddr = addr
-                netmask = ifaceobj.get_attr_value_n('netmask', addr_index)
-                if netmask:
-                    prefixlen = IPNetwork('%s' %addr +
-                                '/%s' %netmask).prefixlen
-                    newaddr = addr + '/%s' %prefixlen
                 else:
-                    # we are here because there is no slash (/xx) and no netmask
-                    # just let IPNetwork handle the ipv4 or ipv6 address mask
-                    prefixlen = IPNetwork(addr).prefixlen
-                    newaddr = addr + '/%s' %prefixlen
-                newaddrs.append(newaddr)
+                    netmask = ifaceobj.get_attr_value_n('netmask', addr_index)
+                    if netmask:
+                        prefixlen = IPNetwork('%s' %addr +
+                                    '/%s' %netmask).prefixlen
+                        newaddr = addr + '/%s' %prefixlen
+                    else:
+                        # we are here because there is no slash (/xx) and no netmask
+                        # just let IPNetwork handle the ipv4 or ipv6 address mask
+                        prefixlen = IPNetwork(addr).prefixlen
+                        newaddr = addr + '/%s' %prefixlen
+                    newaddrs.append(newaddr)
```

Comment:
Where is the netmask dervied? When there's a '/' in the addr, newaddr is correcty set, appended to the newaddrs list and NOT continue'd (lite before). The else part is the "logical equivalent" for the "continue" used before. The "continue" left the loop before the pointopoint handling code was reached so it didn't work with prefixes written as <ip/mask> but only with prefixes configured as <address> and separate <netmask>.

Does this clarify the problem (and solution)?